config/docs-json.config.js

```diff
@@ -46,7 +47,52 @@ StyleDictionary.registerFormat({
  *   comment: '...'
  * },
  */
-function getDocsShape() {
+
+ var levels = {
+	"fail": {
+		range: [0, 3],
+		color: "hsl(0, 100%, 40%)"
+	},
+	"aa-large": {
+		range: [3, 4.5],
+		color: "hsl(40, 100%, 45%)"
+	},
+	"aa": {
+		range: [4.5, 7],
+		color: "hsl(80, 60%, 45%)"
+	},
+	"aaa": {
+		range: [7, 22],
+		color: "hsl(95, 60%, 41%)"
+	}
+};
+
+function getLevel(ratio) {
+  for (const [key, value] of Object.entries(levels)) {
+    if (ratio > value.range[0] && ratio < value.range[1]) {
+      return key
+    }
+  }  
+}
+
+function getContrastCheck(value, element, allTokens) {
+  const format = element.replaceAll('.', '-').replace('&', 'and')
+  const withPrefix = 'telekom-' + format
+  const currentToken = allTokens.find(el => el.name === withPrefix)
+  let contrastRatio;
+  if (currentToken && currentToken.value) {
+    contrastRatio = hex(value, currentToken.value)
```

Comment:
We have values different than hex (e.g. `hsla(0, 0%, 0%, 0.65)` from Text & Icon / Additional). You can probably use tinycolor2, already in the repo, to make sure values passed into the contrast check are always hex (`tinycolor(value).toHexString()`)

Reply:
added that, hope it's fine

Reply:
there's probably better options for these, but I think it's good enough (you know what they say about "naming things" 😄)